### ISSUE TITLE
ADPCM decode doesn't optionally remove sync frames

### ISSUE THREAD

**Issue by dh1df** [jketterl/csdr#9]
There seems to be no counterpart to
`csdr adpcm -e -s`
Just using
`csdr adpcm -d`
creates distortions in the audio from the sync frames and
`csdr adpcm -d -s`
doesn't improve this, the sync flag is ignored in the adpcm decoder.

**Comment by jketterl** [comment]
This is mostly due to the fact that the adpcm decoder does not currently implement the sync mode. I never really go to the part of adding that.

The sync mode was conceived as a way to support late entry, i.e. the ability to start decoding a stream of data at any given entry point (as opposed to right from the beginning). What exactly are you using this feature for?

**Comment by dh1df** [comment]
I have written a python script which gets a adpcm stream from OpenWebRX for further processing (i.e. recording and uploading). However there seems to be no simple way to get rid of the sync frames, they distort the audio. I have hacked together a small program which removes them, but it is slow and horrible :-) Curiously and off-topic also the sample rate has issues, I have requested 11025 and seem to be getting something like 10144. All worked well on an older OpenWebRX version without sync frames

**Comment by jketterl** [comment]
Yeah sorry but I can only repeat this whenever a use case like this comes up: The OpenWebRX websockets are considered internal. They are subject to change. They're not meant to be consumed by third-party processes...

**Comment by dh1df** [comment]
So is there an official API?

**Comment by jketterl** [comment]
No, there isn't. OpenWebRX is not a multi-purpose streaming service that is intended to feed third-party applications. You may need to look into a different project for that purpose. Or maybe wait until the project matures to that degree, but I can't give you any guarantees that that will happen. It is not on the roadmap for now.
